components/brave_wallet/common/eth_request_helper.cc

```diff
@@ -528,12 +528,14 @@ mojom::EthSignTypedDataPtr ParseEthSignTypedDataParams(
     result->meta = nullptr;
   }
 
-  result->domain_hash = domain_hash->first;
+  result->domain_hash.assign(domain_hash->first.begin(),
+                             domain_hash->first.end());
```

Comment:
By the way, these comments I made is because I had overlooked the fact these were arrays. `assign` was fine before, so it is your choice whatever you prefer.